components/script/dom/node.rs

```diff
@@ -349,7 +354,15 @@ impl Node {
             kid.teardown();
         }
     }
+    /// method on Node that returns Vec<Root<MutationObserver>>
+    pub fn registered_mutation_observers(&self) -> &DOMRefCell<Vec<JS<MutationObserver>>> {
```

Comment:
It's difficult to make sure the dynamic checks in borrowing a `DOMRefCell` succeed if you let the cell escape like this. Better(?) to move any functionality which depends on the mutation observers into `Node`?